operations/agent.go

```diff
@@ -113,5 +118,19 @@ func Agent() cli.Command {
 			return err
 		},
 	}
+}
+
+func hardShutdownForSignals(ctx context.Context, serviceCanceler context.CancelFunc) {
+	defer recovery.LogStackTraceAndExit("agent signal handler")
+	sigChan := make(chan os.Signal, 2)
+	signal.Notify(sigChan, syscall.SIGTERM)
 
+	select {
+	case <-ctx.Done():
+	case <-sigChan:
+		grip.Info("service exiting after receiving signal")
+	}
+	serviceCanceler()
+	time.Sleep(time.Second)
```

Comment:
This seems unnecessary. If we're exiting, why wait to let canceled contexts gracefully stop goroutines? Similarly, why pass a context to this function at all? We presumably only want to exit if we receive a SIGTERM. If something else calls cancel, do we really want to `os.Exit`?